src/NServiceBus.Core/Transports/Receiving.cs

```diff
@@ -15,17 +15,24 @@ internal Receiving()
             Defaults(s =>
             {
                 var transportInfrastructure = s.Get<TransportInfrastructure>();
-                var userDiscriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
-                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName") ?? s.EndpointName();
-                if (userDiscriminator != null)
+                var discriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
+                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName");
+                var mainInstance = new EndpointInstance(s.EndpointName());
+                if (baseQueueName != null)
                 {
-                    var p = s.Get<TransportInfrastructure>().BindToLocalEndpoint(new EndpointInstance(baseQueueName, userDiscriminator));
-                    s.SetDefault("NServiceBus.EndpointSpecificQueue", transportInfrastructure.ToTransportAddress(new LogicalAddress(p)));
+                    mainInstance = mainInstance.SetProperty("queue", baseQueueName);
```

Comment:
That's not MSMQ concept. I've tried limiting the implementation of `queue` to MSMQ only but it results in inconsistency where the current endpoint transport address is generated in a different way than other endpoints. Decided to move this here and just not expose it in different transports for the outgoing messages, only for the local endpoint.

This way we can keep the logical stuff well separated from the physical stuff.